tests/test_server.py

```diff
@@ -17,117 +41,284 @@ def get_app(self):
+            '_id': ID,
+            'login': LOGIN,
+            'token': TOKEN,
+            'name': 'STXNext',
+        })
+    )
+    @tornado.testing.gen_test(timeout=100000)
+    def test_new_game_exceptions(self, user_get):
+
+        def new_room(body):
+            return self.client.fetch(
+                self.get_url('/games?token={}'.format(TOKEN)),
+                method='POST',
+                body=json.dumps(body),
+            )
 
-        self.wait()
-
-    def test_admin(self):
-        with unittest.mock.patch('server.User', autospec=True) as user:
-            user.configure_mock(**{
-                'get.return_value': user,
-                'admin': False,
-            })
-
-            response = self.fetch('/games/2', method='PUT', body="")
-
-            self.assertEqual(response.code, http.client.FORBIDDEN)
-
-        with unittest.mock.patch('server.User', autospec=True) as user:
-            user.configure_mock(**{
-                'get.return_value': user,
-                'admin': True,
+        bad_params = [
+            {},  # empty post data
+            {'title': 'too large', 'board_size': 15},
+            {'title': 'too small', 'board_size': 1},
+            {'title': 'wrong type', 'board_size': '1'},
+            {'title': 't'},  # too short
+            {'title': 't'.join([str(x) for x in range(0, 101)])},  # too long
+            {'title': 'duplicated title'},
+        ]
+
+        for params in bad_params:
+            with self.assertRaises(tornado.httpclient.HTTPError):
+                response = yield new_room(params)
+                self.assertEqual(response.code, 404)
+
+        server.game_rooms['5'] = GameRoom(author=LOGIN)
+        with self.assertRaises(tornado.httpclient.HTTPError):
+            response = yield new_room({
+                'title': 'rooms limit reached'
             })
+            self.assertEqual(response.code, 404)
 
-            response = self.fetch('/games/2', method='PUT', body="")
-
-            self.assertEqual(response.code, http.client.OK)
-
-
-class BoardTestCase(GrotTestCase):
-
-    def setUp(self):
-        super(BoardTestCase, self).setUp()
-
-        self.user_patch = unittest.mock.patch('server.User', autospec=True)
-        self.user = self.user_patch.start()
-        self.user.configure_mock(**{
-            'get.return_value': self.user,
-            'id': 1337,
+    @unittest.mock.patch(
+        'server.game_rooms', {
+            ID: GameRoom(
+                _id=ID,
+                author=LOGIN,
+            )
+        }
+    )
+    @unittest.mock.patch(
+        'server.User.collection.find_one',
+        return_value=future_wrap({
+            '_id': ID,
+            'login': LOGIN,
+            'token': TOKEN,
             'name': 'STXNext',
-            'token': '53CR3T',
         })
+    )
+    @tornado.testing.gen_test(timeout=2000000)
+    def test_join_and_play(self, get_user):
+        join_result, start_result = yield [
```

Comment:
I'm wondering how it works. Because now `join_result` will wait forever (because game will not start with just one player).

In this test you should:
1. create room (in `mock.patch`) that allow joining the same player with different aliases (`allow_multi=True`) for two players (`max_players=2`)
2. Add two concurrent (yield with list) calls to `/games/{}/board?...` with different aliases, so the game will start automatically (because room will be full).
3. After start - call '/games/{}' and check is two aliases on list
4. and then POST some moves ('{x: ..., y: ....}') concurrently for both players.
5. check final results
6. delete room

So quite a lot you already have, just few changes.